lib/cisco_node_utils/bgp_af.rb

```diff
@@ -602,11 +635,15 @@ def networks=(should_list)
       [:add, :remove].each do |action|
         Cisco::Logger.debug("networks delta #{@get_args}\n #{action}: " \
                             "#{delta_hash[action]}")
-        delta_hash[action].each do |network, route_map|
+        delta_hash[action].each do |network, route_map_policy|
           state = (action == :add) ? '' : 'no'
           network = Utils.process_network_mask(network)
-          route_map = "route-map #{route_map}" unless route_map.nil?
-          set_args_keys(state: state, network: network, route_map: route_map)
+          unless route_map_policy.nil?
+            route_map = "route-map #{route_map_policy}"
+            route_policy = "route-policy #{route_map_policy}"
+          end
+          set_args_keys(state: state, network: network, route_map: route_map,
+                        route_policy: route_policy)
```

Comment:
My initial idea was to delineate in the setter - create a key string where x = route-policy or route-map and pass x to the yaml. That was an idea from how we'd implemented something similar. The above seems odd to me too.